### PR TITLE
Restructure retry mechanism

### PR DESCRIPTION
**Pull request by ziscky**
ClickUp: [https://app.clickup.com/t/86942wrhv](https://app.clickup.com/t/86942wrhv)

- Wrap the same backoff library used by resty
- Create a method to retry any function ( for use in cases where we can't replace the http client )

:link: [zboto Link](https://app.clickup.com/t/86946az1j)

### CODE REVIEW

**Review by lucaslopezf — CHANGES_REQUESTED** [review]
By setting the backoff this way, we disrupt the httpClient interface as well as the mocks used for testing. We could consider adopting a backoff strategy similar to the one used in Resty (following the builder pattern): https://github.com/go-resty/resty?tab=readme-ov-file#retries. Doing so would align our HTTP client with a standard, making it more intuitive and testable

**Review by emmanuelm41 — APPROVED** [review]
I like the approach. Let's wait for Lucas approval, but it is ok for me!

**Review by lucaslopezf — APPROVED** [review]
lgtm! 💪

**Pull request merged**
